src/commonMain/kotlin/app/cash/turbine/channel.kt

```diff
@@ -64,16 +70,44 @@ public fun <T> ReceiveChannel<T>.expectNoEvents() {
  *
  * This function will always return a terminal event on a closed [ReceiveChannel].
  */
-public suspend fun <T> ReceiveChannel<T>.awaitEvent(): Event<T> =
-  try {
-    Event.Item(receive())
+@OptIn(ExperimentalCoroutinesApi::class)
+public suspend fun <T> ReceiveChannel<T>.awaitEvent(): Event<T> {
+  val timeoutMs = contextTimeout()
+  val testScheduler = coroutineContext[TestCoroutineScheduler]
+  return try {
+    withTimeout(timeoutMs) {
+      val item = if (testScheduler == null) {
+        // With no test scheduler, let receive() expire the timeout. This will use wallclock time.
+        receive()
+      } else {
+        // *With* a test scheduler, we must poll and nudge the clock
+        // until some kind of result is produced.
```

Comment:
Not sure about this. Before removing timeout, I changed it to _always_ be wallclock time and _never_ dispatcher time. If you want to assert a timeout on the dispatcher's fake time then you can just write `withTimeout(1.seconds) { awaitItem() }`. Turbine's timeout was always meant to be a hang detector. I don't like that this makes the behavior subtle and reliant on factors which are hard to document and hard to discern when glancing at a test.

Reply:
Two points need to be responded to: (1) whether timeouts have a role other than bailing out hung code, and (2) whether `withTimeout(1.seconds)` is sufficient as an implementation.

1. Hangs are garden variety error conditions in CSP-style concurrent code; timeouts at `receive` sites provide actionable feedback on why these hangs occur.

Here's an example of a test that fails by hanging:

```kotlin
@Test fun logsSubmitClicked() = runTest {
  val presenter = Presenter(fakeAnalytics, fakeAppService)
  fakeAppService.responses += ApiResult(Success("What's new today"))
  presenter.models(events).test {
    assertThat(awaitItem()).isEqualTo(Loading)
    events.emit(SubmiClicked())
    assertThat(fakeAnalytics.events.awaitItem()).isEqualTo(SubmitClickAnalyticsEvent())
    assertThat(awaitItem()).isEqualTo(Content("What's new today"))
  }
}
```

Some failure modes for this test:

* `Loading` isn't emitted first, but skips straight to `Content`
* Incorrect `Content`
* Analytics event is never emitted
* `Content` is never emitted

Half of those will present as a hung test. Seeing exactly which one of these happened in CI is invaluable.

2. I had hoped that this:
```kotlin
runTest {
  flow.test {
    assertFailsWith<TimeoutException> { withTimeout(1.seconds) { awaitEvent() } }
  }
}
```

...would work with our new `runTest`. I initially wrote it that way, and unfortunately it does not: `withTimeout(1.seconds)` requires a concurrent coroutine to tick the virtual clock in order to advance. This makes it useless as a testing API.

> I don't like that this makes the behavior subtle and reliant on factors which are hard to document and hard to discern when glancing at a test.

The `delay(timeoutMs / 10)` is the real bugaboo that needs answering. 

I am able to convince myself that automatic manipulation of virtual time is worth it, because 

1. It speeds iteration on failing tests that timeout
2. it makes it possible to implement `awaitNoEvents()`. `awaitNoEvents()` is something colleagues have asked for that solves some real testing problems, and it can't realistically be done on wall clock time. (it'd have to do something hacky like `repeat(10) { yield() }; expectNoEvents()`) My answer thus far has been, "Tough, you can't logically assert that nothing happens in response to a stimulus," but it sure would be nice to actually solve the problem instead being a difficult theory nerd.

Implicitly advancing the clock is a pretty hard nut to swallow, though, and `delay(timeout / 10)` even more so. `delay(timeout / 10)` can probably be rewritten to be `delay(10)` and solve the problem without being too confusing, but if you think that implicitly advancing virtual time will cause too much confusion to be worth the advantages above, then this approach should be ditched in favor of wall clock time.

Reply:
Where do we want to land on this? If you're not comfortable with the virtual time thing, I would like to get wall clock merged as soon as possible. Without it, I don't know how to tell people how to use 0.9. I keep on hacking around it myself.

Re: details of wall clock time, I am concerned about the need to use `Dispatchers.Default`. It doesn't feel safe. Even if we did something like this:

```kotlin
  val callingContext = currentCoroutineContext()
  withContext(Dispatchers.Default) {
    withTimeout(timeoutMs) {
      withContext(callingContext) {
        receive()
      }
    }
  }
```

...having multiple threads involved makes me antsy.